### PR TITLE
feat(l1): initiate rlpx connections from discv4

### LINKED ISSUE #837: Initiate RLPx connections from discv4
Currently discv4 handles the peer discovery, and it has to delegate the RLPx connection to a new task.

See [this TODO](https://github.com/lambdaclass/lambda_ethereum_rust/pull/716/files#diff-db398f56bc535b9d6cf5fdfcbfceafafd7f34fb6825731c4c6781c0ca6d15764R182-R191) and remove [this harcoded test](https://github.com/lambdaclass/lambda_ethereum_rust/pull/716/files#diff-db398f56bc535b9d6cf5fdfcbfceafafd7f34fb6825731c4c6781c0ca6d15764R50) and [it's related function](https://github.com/lambdaclass/lambda_ethereum_rust/pull/716/files#diff-db398f56bc535b9d6cf5fdfcbfceafafd7f34fb6825731c4c6781c0ca6d15764R713)

### PR DESCRIPTION
**Pull request by ElFantasma**
**Motivation**

DiscV4 protocol discovers new peers, and after deciding it is a valid peer, a TCP connection is established and RLPx protocol starts

**Description**

Now, when receiving a DiscV4 Pong message and after evaluating the peer, a RLPx connection is created and established. If handshake fails it removes the peer from the Kademlia table.

Closes #837.

Also removed some hard-coded testing code.

Once the listen loop is built (#840) there may be other conditions where a peer has to be discarded. (eg. when the other peer sends a Disconnect message). After #840 is completed we may create some more issues on this regard.

### CODE REVIEW

**Review by fkrause98 — COMMENTED** [review]
Left some comments, is there a proper way I can test this yet?

File: crates/networking/p2p/net.rs [hunk 1]
```diff
@@ -148,7 +160,8 @@ async fn discover_peers_server(
                         table.insert_node(Node {
                             ip: from.ip(),
                             udp_port: from.port(),
-                            tcp_port: 0,
+                            // TODO: Check how to obtain proper tcp port
```
[the remark this answers is not in the document]
Comment (ElFantasma) [reply crates/networking/p2p/net.rs:163]: Not really, but its related to https://github.com/lambdaclass/lambda_ethereum_rust/issues/905 

I will create a ticket, it's more a matter of researching if it makes sense to know the tcp port at this place and how to obtain it.

File: crates/networking/p2p/net.rs [hunk 2]
```diff
@@ -180,17 +194,14 @@ async fn discover_peers_server(
...
-                        let mut msg_buf = vec![0; read - (32 + 65)];
-                        buf[32 + 65..read].clone_into(&mut msg_buf);
-                        let mut sig_bytes = vec![0; 65];
-                        buf[32..32 + 65].clone_into(&mut sig_bytes);
-                        let _signer_clone = signer.clone();
-                        // tokio::spawn(async move {
-                        //     handle_peer_as_initiator(signer_clone, &msg_buf, &node).await;
-                        // });
+
+                        let mut msg_buf = vec![0; read - 32];
```
[the remark this answers is not in the document]
Comment (ElFantasma) [reply crates/networking/p2p/net.rs:198]: Yeah, there is a ticket to tackle all the error handling [here](https://github.com/lambdaclass/lambda_ethereum_rust/issues/845).

I left that out at this stage to move forward

File: crates/networking/p2p/rlpx/connection.rs [hunk 3]
```diff
@@ -289,6 +289,14 @@ impl<S: AsyncWrite + AsyncRead + std::marker::Unpin> RLPxConnection<S> {
             _ => panic!("Received an unexpected message"),
         }
     }
+
+    pub fn get_remote_node_id(&self) -> H512 {
+        match &self.state {
+            RLPxConnectionState::Established(state) => state.remote_node_id,
+            // TODO proper error
+            _ => panic!("Invalid state"),
```
[the remark this answers is not in the document]
Comment (ElFantasma) [reply crates/networking/p2p/rlpx/connection.rs:297]: All functions relying on state should be returning `Result<>` to be able to handle different error conditions. Left a separated [ticket](https://github.com/lambdaclass/lambda_ethereum_rust/issues/845) for that

File: crates/networking/p2p/net.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ElFantasma) [reply crates/networking/p2p/net.rs]: Actually, removed this couple of lines. `tcp_port: 0` was ok at this place as we don't yet know the tcp_port (we are only communicating via udp).

Proper tcp port configuration should be addressed in https://github.com/lambdaclass/lambda_ethereum_rust/issues/905

**Comment by ElFantasma** [comment]
> Left some comments, is there a proper way I can test this yet?

I'm testing it locally setting up a geth node and connecting it with ours. Eventually this will be tested on [hive tests that test the `eth` capability](https://github.com/lambdaclass/lambda_ethereum_rust/issues/931) (couldn't find tests for `p2p` capability)

**Review by fkrause98 — APPROVED** [review]
LGTM, 
My biggest concert is proper error handling, but we'll address it in future issues.

**Pull request merged**